tests/unit_tests/pyanaconda_tests/modules/subscription/test_satellite.py

```diff
@@ -0,0 +1,233 @@
+        script_text = download_satellite_provisioning_script("https://satellite.example.com")
+        # check script text was returned
+        self.assertEqual("foo script text", script_text)
+        # check the session was called correctly
+        session.get.assert_called_once_with(
+            'https://satellite.example.com' + PROVISIONING_SCRIPT_SUB_PATH,
+            headers={"user-agent": USER_AGENT},
+            proxies={},
+            verify=False,
+            timeout=NETWORK_CONNECTION_TIMEOUT
+        )
+        result.close.assert_called_once()
+
+    @patch("pyanaconda.core.util.requests_session")
+    def script_download_not_ok_test(self, get_session):
+        """Test the download_satellite_provisioning_script function - result not ok."""
+        # mock the Python Request session
+        session = get_session.return_value.__enter__.return_value
+        result = session.get.return_value
+        result.ok = False
+        result.text = "foo script text"
+        # run the download method
+        script_text = download_satellite_provisioning_script("satellite.example.com")
+        # if result has ok == False, None should be returned instead of script text
+        self.assertIsNone(script_text)
+        # check the session was called correctly
+        session.get.assert_called_once_with(
+            'http://satellite.example.com' + PROVISIONING_SCRIPT_SUB_PATH,
+            headers={"user-agent": USER_AGENT},
+            proxies={},
+            verify=False,
+            timeout=NETWORK_CONNECTION_TIMEOUT
+        )
+        result.close.assert_called_once()
+
+    @patch("pyanaconda.core.util.requests_session")
+    def script_download_exception_test(self, get_session):
+        """Test the download_satellite_provisioning_script function - exception."""
+        # mock the Python Request session
+        session = get_session.return_value.__enter__.return_value
+        session.get.side_effect = RequestException()
+        # run the download method
+        script_text = download_satellite_provisioning_script("satellite.example.com")
+        # if requests throw an exception, None should be returned instead of script text
+        self.assertIsNone(script_text)
+        # check the session was called correctly
+        session.get.assert_called_once_with(
+            'http://satellite.example.com' + PROVISIONING_SCRIPT_SUB_PATH,
+            headers={"user-agent": USER_AGENT},
+            proxies={},
+            verify=False,
+            timeout=NETWORK_CONNECTION_TIMEOUT
+        )
+
+    def run_satellite_provisioning_script_no_script_test(self):
+        """Test the run_satellite_provisioning_script function - no script."""
+        # if no script is provided, False should be returned
+        self.assertFalse(run_satellite_provisioning_script(provisioning_script=None))
+
+    @patch('pyanaconda.modules.subscription.satellite.util.mkdirChain')
+    @patch('tempfile.NamedTemporaryFile')
```

Comment:
You should be able to test `run_satellite_provisioning_script` for real without any mocking. We usually use `with tempfile.TemporaryDirectory() as sysroot:` to create a sysroot for testing.